### DISCUSSION TITLE
Enable compression in DGS Framework

### DISCUSSION THREAD

**Discussion by saravind-hotstar** [Netflix/dgs-framework#1310] [Q&A]
Is there a way to enable compression in DGS framework

**Comment by srinivasankavitha** [comment]
We don't have any explicit support for enabling compression of response payloads.

**Reply by saravind-hotstar** [reply]
Apollo router now supports request and response compression 

- https://www.apollographql.com/docs/router/configuration/traffic-shaping/  to subgraphs in federation spec
- https://github.com/apollographql/router/pull/1229

Is there any way in DGS this can be enabled, via spring

**Comment by srinivasankavitha** [comment]
We don't have support in the DGS framework for this yet, and I'm not aware
of a mechanism to achieve this in spring out of the box. We will look into
adding it in the future, but for now it is not available.

**Comment by berngp** [comment]
@saravind-hotstar, Spring Boot already supports [HTTP Response Compression]. Are you looking for something different?

Ref [server.server.compression Properties].


[HTTP Response Compression]: https://docs.spring.io/spring-boot/docs/2.7.x/reference/htmlsingle/#howto.webserver.enable-response-compression
[server.server.compression Properties]: https://docs.spring.io/spring-boot/docs/2.7.x/reference/htmlsingle/#application-properties.server.server.compression.enabled

**Reply by saravind-hotstar** [reply]
@berngp Thanks, we were looking for compression in both ways, i.e request and response. In case we wanted to enable request compression, spring does support filters, is this the way to go forward. Any specific DGS entry points to look about.

**Comment by berngp** [comment]
@saravind-hotstar the DGS framework is built on top of Spring Boot and holds no opinion on which embedded servlet container you use. I think you should look into enabling request and response compression depending on the one you use. I think all of them should support response compression but I am not sure which, for example Tomcat, supports request deflation.
